include/LeMonADE/updater/UpdaterAbstractCreate.h

```diff
@@ -0,0 +1,301 @@
+    move_system(2);
+    counter++;
+  }
+  return false;
+}
+
+/******************************************************************************/
+/**
+ * @brief function to add a monomer to a specific position. if position is not free return false 
+ * @param VectorInt3 position
+ * @param type attribute tag of the new monomer
+ * @return <b false> if position is not free, <b true> if move was applied
+ */
+template<class IngredientsType>
+bool UpdaterAbstractCreate<IngredientsType>::add_monomer_to_position(VectorInt3 position, int32_t type){
+  MoveAddScMonomer addmove;
+  addmove.init(ingredients);
+  addmove.setPosition(position);
+  if(addmove.check(ingredients)==true){
+    addmove.setType(type);
+    addmove.apply(ingredients);
+    return true;
+  }else{
+    return false;
+  }
+  
+}
+
+/******************************************************************************/
+/**
+ * @brief function to add a new monomer between two already existing ones instead of a bond.
+ * @param indexA
+ * @param indexB
+ * @param type attribute tag of the new monomer
+ * @return <b false> if position is not free, <b true> if move was applied
+ */
+template<class IngredientsType>
+bool UpdaterAbstractCreate<IngredientsType>::add_monomer_inside_connected_pair(uint32_t indexA, uint32_t indexB, int32_t type){
+  //first check if monomers are connected
+  if( ! ingredients.getMolecules().areConnected(indexA,indexB))
+    return false;
+  
+  MoveAddScMonomer addmove;
+  addmove.init(ingredients);
+  
+  int32_t counter(0);
+  
+  while(counter<10000){
+    //try at most 30 random bondvectors to find a new monomer position
+    for(uint i=0;i<30;i++){
+      // get id of random bondvector with index <= 22, also P(2,0,0)
+      std::size_t randBondVectorID((rng.r250_rand32() % 6)+17);
+      VectorInt3 bondvector(ingredients.getBondset().getBondVector(randBondVectorID));
+      // set position of new monomer
+      addmove.setPosition(ingredients.getMolecules()[indexA]+bondvector);
+    
+      // check new position (excluded volume, other features)
+      if(addmove.check(ingredients)==true){
+	//check the new bondvector bewten the new monomer and indexB
+	VectorInt3 checkBV(addmove.getPosition()-ingredients.getMolecules()[indexB]);
+	if( (checkBV.getLength() < 3) && (ingredients.getBondset().isValidStrongCheck(checkBV)) ){
+	  addmove.setType(type);
+	  addmove.apply(ingredients);
+	  ingredients.modifyMolecules().connect( indexA, (ingredients.getMolecules().size()-1) );
+	  ingredients.modifyMolecules().connect( indexB, (ingredients.getMolecules().size()-1) );
+	  ingredients.modifyMolecules().disconnect( indexA, indexB );
+	  return true;
+	}
+      }
+    }
+    // if no position matches, we need to move the system a bit
+    move_system(2);
+    counter++;
+  }
+  // create a meaningful return message
+  std::cout << "UpdaterAbstractCreate::add_monomer_inside_connected_pair:  could not add new monomer between "<<indexA<< " and "<<indexB<<std::endl; 
```

Comment:
I would suggest to handle errors consistently. Either we throw exceptions, or give error messages here in all add-functions, in case stuff goes wrong, or we don't do it at all. In the end, the user knows anyways that it did not work, because the function returns false. 
Thus my suggestion: delete the errormessage.